### DISCUSSION TITLE
“rabbitmq_delayed_message_exchange” implementation in C++ AMQPClient

### DISCUSSION THREAD

**Discussion by woznez** [rabbitmq/rabbitmq-delayed-message-exchange#183] [Q&A]
I am trying to implement “rabbitmq_delayed_message_exchange” in C++ but I did not come up with an example I found implementation in java, python but did not find any implementation in C++. If anyone can help me on this.

**Comment by lukebakken** [comment]
> I am trying to implement “rabbitmq_delayed_message_exchange” in C++ but I did not come up with an example I found implementation in java, python

Please provide links to the examples in Java and / or Python that you are referring to.

Which C++ library are you using? What have you already tried?

**Reply by woznez** [reply]
Hello,
  The link i am referring to : 
https://blog.rabbitmq.com/posts/2015/04/scheduling-messages-with-rabbitmq
I am using SimpleAmqpClient library and i am not getting any API reference for c++

**Reply by lukebakken** [reply]
I suggest asking the project's maintainers - 

https://github.com/alanxz/SimpleAmqpClient

Basically you need to declare the Exchange correctly (all of this is Java):

```java
channel.exchangeDeclare("my-exchange", "x-delayed-message", true, false, args);
```

Then publish to the exchange using the correct header:

```java
headers.put("x-delay", 5000)
```

I don't have time to figure this out myself. What I can do, however, is _assist_ you if you provide code I can clone, compile, and run.
